src/Sarif.Driver.UnitTests/Sdk/FormatForVisualStudioTests.cs

```diff
@@ -0,0 +1,161 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using FluentAssertions;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Sarif.Driver.Sdk
+{
+    // These tests test the extension method Result.FormatForVisualStudio.
+    // But by providing various Region objects and ResultKind values, they
+    // also exercise Region.FormatForVisualStudio and ResultKind.FormatForVisualStudio.
+    public class FormatForVisualStudioTests
+    {
+        private const string TestRuleId = "TST0001";
+        private const string TestFormatSpecifier = "testFormatSpecifier";
+        private const string TestAnalysisTarget = @"C:\dir\file";
+        private static readonly string DisplayedTarget = TestAnalysisTarget.Replace('\\', '/');
+
+        private static readonly RuleDescriptor TestRule = new RuleDescriptor(
+            TestRuleId,
+            "ThisIsATest",
+            "short description",
+            "full description",
+            null,           // options
+            new Dictionary<string, string>
+            {
+                [TestFormatSpecifier] = "First: {0}, Second: {1}"
+            },
+            null,           // helpUri
+            null,           // properties
+            null);          // tags
+
+        private static readonly Region MultiLineTestRegion = new Region
```

Comment:
There are missing cases for formatting the region. Will these go in tests associated with the code that produces the (x,x) notation?